stacks-signer/src/runloop.rs

```diff
@@ -68,60 +50,47 @@ pub struct RunLoop<C> {
-                }
-                self.state = State::Dkg;
-            }
-            (RunLoopCommand::Dkg, State::DkgDone) => {
-                info!("Done DKG");
-                self.state = State::Exit;
-            }
-            (RunLoopCommand::DkgSign { message }, State::Idle) => {
-                info!("Starting DKG for signing message: {:?}", message);
-                if let Ok(msg) = self.coordinator.start_distributed_key_generation() {
-                    let _ = self
-                        .stacks_client
-                        .send_message(self.signing_round.signer.signer_id, msg);
-                }
-                self.state = State::Dkg;
-            }
-            (RunLoopCommand::DkgSign { message }, State::DkgDone)
-            | (RunLoopCommand::Sign { message }, State::Idle) => {
-                info!("Signing message: {:?}", message);
-                if let Ok(msg) = self.coordinator.start_signing_message(message) {
-                    let _ = self
-                        .stacks_client
-                        .send_message(self.signing_round.signer.signer_id, msg);
+    fn execute_command(&mut self) {
+        match self.state {
+            State::Idle => {
+                if let Some(command) = self.commands.pop_front() {
+                    match command {
+                        RunLoopCommand::Dkg => {
+                            info!("Starting DKG");
+                            if let Ok(msg) = self.coordinator.start_distributed_key_generation() {
+                                let ack = self
+                                    .stacks_client
+                                    .send_message(self.signing_round.signer.signer_id, msg);
+                                debug!("ACK: {:?}", ack);
+                            }
+                            self.state = State::Dkg;
+                        }
+                        RunLoopCommand::Sign { message } => {
+                            info!("Signing message: {:?}", message);
+                            if let Ok(msg) = self.coordinator.start_signing_message(&message) {
+                                let _ = self
+                                    .stacks_client
+                                    .send_message(self.signing_round.signer.signer_id, msg);
+                            }
+                            self.state = State::Sign
+                        }
+                    }
                 }
-                self.state = State::Sign
             }
-            (RunLoopCommand::DkgSign { message }, State::SignDone)
-            | (RunLoopCommand::Sign { message }, State::SignDone) => {
-                info!("Done signing message: {:?}", message);
-                self.state = State::Exit;
+            State::Dkg | State::Sign => {
+                // We cannot execute the next command until the current one is finished...
+                // Do nothing
+                debug!("Waiting for operation to finish");
```

Comment:
Does this command just get dropped then?  Asking because the signer daemon will receive a list of things it needs to sign (sBTC transactions, block headers), and it can't just drop them.  Will these requests be buffered up somewhere?

Reply:
It gets stored in the command queue until it is able to process the next one. This will have to be reviewed down the line to enable processing multiple things at once, but currently it does it sequentially, buffering commands as they are received until it can continue processing them.